src/main/java/com/google/api/generator/engine/ast/OperatorKind.java

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+public enum OperatorKind {
+  ARITHMETIC_ADDITION(true),
+  RELATIONAL_EQUAL_TO(true),
+  RELATIONAL_NOT_EQUAL_TO(true),
+  UNARY_LOGICAL_NOT(false),
+  UNARY_POST_INCREMENT(true);
+
+  private final boolean isPostfixOperator;
+
+  OperatorKind(final boolean isPostfixOperator) {
+    this.isPostfixOperator = isPostfixOperator;
+  }
+
+  public boolean isPostfixOperator() {
+    return this.isPostfixOperator;
```

Comment:
Instead of passing in a boolean to each enum, what about just doing `return !this.equals(UNARY_LOGICAL_NOT)`?

Reply:
Also, consider whether the two-expression operators should return true or false here.

Reply:
Got your point. Had a little think on this, prefix operators are(!) (++) (--) (+) (-), instead of use this.equals(UNARY_LOGICAL_NOT) && this.equals(UNARY_PREFIX_INCREMENT)...
I added a enumset for looking up the operator.
LMK if you have concern. Thanks.